src/elements/StoryBookmark/StoryBookmark.tsx

```diff
@@ -0,0 +1,78 @@
+import { StoryBookmarkNode, STORY_BOOKMARK_NODE_TYPE } from '@prezly/slate-types';
+import { StoryBookmarkLayout } from '@prezly/slate-types';
+import React, { useRef, useState, useMemo } from 'react';
+
+import { useResizeObserver, utils } from '../../lib';
+
+import { BookmarkCard } from '../../components';
+import { useElementContext } from '../../RendererContext';
+
+interface StoryBookmarkBlockProps {
+    node: StoryBookmarkNode;
+}
+
+const HORIZONTAL_LAYOUT_MIN_WIDTH = 480;
+
+export function StoryBookmark({ node }: StoryBookmarkBlockProps) {
+    const ctx = useElementContext(STORY_BOOKMARK_NODE_TYPE);
+    const card = useRef<HTMLDivElement | null>(null);
+    const [isSmallViewport, setSmallViewport] = useState(false);
+
+    useResizeObserver(card.current, function (entries) {
+        entries.forEach(function (entry) {
+            setSmallViewport(entry.contentRect.width < HORIZONTAL_LAYOUT_MIN_WIDTH);
+        });
+    });
```

Comment:
Not a fan of this. In an ideal world, this should be encapsulated into the `BookmarkCard` component, so the consuming element renderers do not need to duplicate this logic.

I see a solution in refactoring `BookmarkCard` into a black-box component when all the details are passed into it as plain properties. With that approach, `BookmarkCard` will not only be easier to use (no need to remember which components in which order should be composed), but also it can encapsulate this responsiveness logic. '

```tsx
<BookmarkCard
  layout={layoout}
  url={...}
  title={...}
  description={...}
  thumbnail={{ url: ..., width: ..., height: ... }}
  provider={{ url: ..., name: ... }}
/>

```